File: mobile-app/lib/models/avis.dart
class Avis {
  final int id;
  final int produitId;
  final String auteur;
  final String commentaire;
  final int note;
  final DateTime? dateCreation;

  Avis({
    required this.id,
    required this.produitId,
    required this.auteur,
    required this.commentaire,
    required this.note,
    this.dateCreation,
  });

  factory Avis.fromJson(Map<String, dynamic> json) {
    return Avis(
      id: json['id'],
      produitId: json['produitId'],
      auteur: json['auteur'],
      commentaire: json['commentaire'] ?? '',
      note: json['note'],
      dateCreation: json['dateCreation'] != null
          ? DateTime.parse(json['dateCreation'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'produitId': produitId,
      'auteur': auteur,
      'commentaire': commentaire,
      'note': note,
    };
  }
}

File: avis-service/src/main/java/com/boutique/avis/entity/Avis.java
package com.boutique.avis.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "avis", indexes = {
    @Index(name = "idx_produit_id", columnList = "produitId")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Avis {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private Long produitId;
    
    @Column(nullable = false, length = 100)
    private String auteur;
    
    @Column(length = 1000)
    private String commentaire;
    
    @Column(nullable = false)
    private Integer note; // 1 à 5
    
    private LocalDateTime dateCreation;
    
    @PrePersist
    protected void onCreate() {
        dateCreation = LocalDateTime.now();
    }
}
